test/expr/primary/keypath/keypath-objc.swift

```diff
@@ -150,4 +150,17 @@ class SR_10146_1 {
 
 class SR_10146_2: SR_10146_1 {
   let a = \AnyObject.b // expected-error {{invalid component of Swift key path}}
-}
\ No newline at end of file
+}
+
+class SR_10146_3 {
+  @objc let abc: Int = 1
+  
+  func doNotCrash() {
+    let _: KeyPath<AnyObject, Int> = \.abc // expected-error {{invalid component of Swift key path}}
+  }
+
+  func doNotCrash_1(_ obj: AnyObject, _ kp: KeyPath<AnyObject, Int>) {
+    let _ = obj[keyPath: \.abc] // expected-error {{invalid component of Swift key path}}
+    let _ = obj[keyPath: kp]
```

Comment:
Question still stands - why is this accepted or it's just missing a diagnostic?

Reply:
I have figured it out - in this case, we call `simplifyKeyPathApplicationConstraint()` and not `simplifyKeyPathConstraint`. I just need to duplicate the code or extract into helper function to fix this

Reply:
I think that’s not really a problem. Since you cannot form a value of type `KeyPath<AnyObject, Int>` it’s effectively uninhabited so the keypath application is not possible to execute. One thing I would try is that SILGen doesn’t crash when generating it though.

Reply:
It's okay, it gets diagnosed now!

Reply:
Yeah, I'm just trying to make sure that we have some erroneous cases covered as well which doen't case a crash in SILGen